dev-util/serve-d/serve-d-0.8.0_beta17.ebuild

```diff
@@ -0,0 +1,154 @@
+# Copyright 2023-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+MY_PV="$(ver_rs 3 - 4 .)" # 0.8.0_beta15 -> 0.8.0-beta.15
+DUB_DEPENDENCIES=(
+	"automem@0.6.9"
+	"cachetools@0.4.1"
+	"dcd@0.16.0-beta.2"
+	"dfmt@0.15.1"
+	"diet-complete@0.0.3"
+	"diet-ng@1.8.1"
+	"dscanner@0.16.0-beta.4"
+	"dub@1.36.0"
+	"emsi_containers@0.9.0"
+	"eventcore@0.9.28"
+	"fuzzymatch@1.0.0"
+	"inifiled@1.3.3"
+	"isfreedesktop@0.1.1"
+	"libasync@0.8.6"
+	"libddoc@0.8.0"
+	"libdparse@0.23.2"
+	"memutils@1.0.10"
+	"mir-algorithm@3.21.0"
+	"mir-core@1.7.0"
+	"mir-cpuid@1.2.11"
+	"mir-ion@2.2.1"
+	"mir-linux-kernel@1.0.1"
+	"msgpack-d@1.0.5"
+	"openssl@3.3.3"
+	"openssl-static@1.0.2+3.0.8"
+	"requests@2.1.3"
+	"rm-rf@0.1.0"
+	"sdlfmt@0.1.1"
+	"sdlite@1.1.2"
+	"serve-d@0.8.0-beta.17"
+	"silly@1.1.1"
+	"standardpaths@0.8.2"
+	"stdx-allocator@2.77.5"
+	"taggedalgebraic@0.11.22"
+	"test_allocator@0.3.4"
+	"unit-threaded@0.10.8"
+	"vibe-container@1.1.0"
+	"vibe-core@2.7.4"
+	"vibe-d@0.9.7"
+	"xdgpaths@0.2.5"
+)
+inherit dub
+
+DUB_TEST_DEP="gitcompatibledubpackage@1.0.4"
+
+DESCRIPTION="Microsoft language server protocol implementation for D"
+HOMEPAGE="https://github.com/Pure-D/serve-d"
+SRC_URI="
+	${DUB_DEPENDENCIES_URIS}
+	test? ( $(dub_dependencies_uris "${DUB_TEST_DEP}") )
+"
+S="${WORKDIR}/${PN}-${MY_PV}"
+LICENSE="MIT BSD"
+LICENSE+=" Apache-2.0 Boost-1.0 BSD GPL-3 LGPL-3 MIT openssl public-domain Unlicense"
+LICENSE+=" test? ( ISC public-domain )" # test dependencies. These do _not_ map to test? () in SRC_URI
```

Comment:
Out of interest, why don't they map? Some of it is bundled I assume?

Reply:
dub is very nice in that it will require that all possible dependencies to be satisfied when building something, even if, in that build, fewer things are needed. In this example, `serve-d` uses `silly` for tests. If `silly` is not present during `dub build` dub will fail. `dub build` does not use silly, only `dub test` so the license of `silly` I've put behind `test?`. There's probably more of these cases, at least both the `openssl` dependencies are unused as well though I didn't want to bother any more with this.

Upstream has an issue for this: https://github.com/dlang/dub/issues/1706